app/src/androidTest/java/com/cmput301f20t21/bookfriends/ui/login/CustomTestRunner.java

```diff
@@ -0,0 +1,151 @@
+package com.cmput301f20t21.bookfriends.ui.login;
+
+import android.app.Application;
+import android.content.Context;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.runner.AndroidJUnitRunner;
+
+import com.cmput301f20t21.bookfriends.BuildConfig;
+
+import org.junit.runner.Description;
+import org.junit.runner.Runner;
+import org.junit.runner.manipulation.Filter;
+import org.junit.runner.manipulation.Filterable;
+import org.junit.runner.manipulation.NoTestsRemainException;
+import org.junit.runner.manipulation.Sortable;
+import org.junit.runner.manipulation.Sorter;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+public final class CustomTestRunner extends Runner implements Filterable, Sortable {
+
+    private static final String TAG = "CustomTestRunner";
+
+    private final Runner delegate;
+
+    public CustomTestRunner(Class<?> klass) throws InitializationError {
+        delegate = loadRunner(klass);
+        BuildConfig.IS_TESTING.set(true);
```

Comment:
I copied this class from `AndroidJUnit4.class`, I wanted to extends it and add line 33 but it didn't let me.

Reply:
question: what's the purpose of a new Test Runner class? were we not able to set the IS_TESTING to false in a normal test case?

Reply:
we can but every test we will need to set it. Having a custom test runner we can abstract that, more convenient (maybe in the future we want to change the way we set IS_TESTING, this approach we be easier to modify).